packages/react-ui/src/widgets/PieWidgetUI/hooks/usePieCategories.js

```diff
@@ -0,0 +1,77 @@
+import { useMemo, useRef, useEffect } from 'react';
+import { useTheme } from '@mui/material';
+import { OTHERS_CATEGORY_NAME, ORDER_TYPES } from '../../utils/chartConstants';
+import { sortDataDescending } from '../../utils/chartUtils';
+
+function usePieCategories(data, order, maxItems, colors) {
+  const theme = useTheme();
+  const colorByCategory = useRef({});
```

Comment:
I know it's old, but just raising awareness.

I wonder if keeping color mapping in ref makes sense. I would say, not.

This makes color mapping stable across data changes (apart from unmounting/remounting that resets ref, but that's other story)

I believe, If we change data we should always apply first color to first (or most dominant) category in _current_ data, not try to keep colors stable.

This is at least how i did it in TS widget.

What do you think?

So

Reply:
Good catch, fixed here https://github.com/CartoDB/carto-react/pull/779/commits/505f6251f853852372ab87c5f83009bb04a8e0ad